src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,239 @@
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def upload_gguf(self) -> None:
+        try:
+            hf_api = HfApi()
+            resp = hf_api.upload_file(
+                path_or_fileobj=self.local_model_path,
+                path_in_repo=self.model,
+                repo_id=self.destination,
+                revision=self.release,
+                token=self.hf_token,
+            )
+            logger.debug(f"Hugging Face response:\n{resp}")
+        except Exception as exc:
+            click.secho(
+                f"\nUploading GGUF model at {self.local_model_path} failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            f"\nUploading GGUF model at {self.local_model_path} succeeded!",
+            fg="green",
+        )
+
+    def upload_safetensors(self) -> None:
+        try:
+            hf_api = HfApi()
+            resp = hf_api.upload_folder(
+                folder_path=self.local_model_path,
+                repo_id=self.destination,
+                revision=self.release,
+                token=self.hf_token,
+            )
+            logger.debug(f"Hugging Face response:\n{resp}")
+        except Exception as exc:
+            click.secho(
+                f"\nUploading safetensors model at {self.local_model_path} failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            f"\nUploading safetensors model at {self.local_model_path} succeeded!",
+            fg="green",
+        )
+
+
+@click.command()
+@click.option(
+    "--model",
+    "-m",
+    required=True,
+    help="Path to the model to upload or name of an existing checkpoint.",
+)
+@click.option(
+    "--dest-type",
+    type=click.Choice(["hf", "oci", "s3"]),
+    default="hf",
+    help="The type of destination to upload to - can be 'hf', 'oci', or 's3'",
+)
+@click.option(
+    "--destination",
+    "-d",
+    type=str,
+    required=True,
+    help="Destination for the model to be uploaded to. Ex: for a Hugging Face repo, should be '<username>/<repo-name>'",
+)
+@click.option(
+    "--release",
+    "-rl",
+    default="main",
+    show_default=True,
+    help="The revision to upload the model to - e.g. a branch for Hugging Face repositories.",
+)
+@click.option(
+    "--hf-token",
+    default="",
+    envvar="HF_TOKEN",
+    help="User access token for connecting to the Hugging Face Hub.",
+)
+@click.pass_context
+@clickext.display_params
+def upload(ctx, model, dest_type, destination, release, hf_token):
+    """Uploads model to a specified location"""
+    uploader = None
+
+    if dest_type == "hf":
```

Comment:
a constant would be good here. `DEFAULTS` perhaps? `DEFAULTS.UPLOAD_PROVIDERS`?

Reply:
I can do a constant for the Click options similar to what we do for Evaluation